src/GHC/Util/Brackets.hs

```diff
@@ -18,6 +21,9 @@ class Brackets a where
   -- | Is the child safe free from brackets in the parent
   -- position. Err on the side of caution, True = don't know.
   needBracket :: Int -> a -> a -> Bool
+  -- | Is a bracket for the child redundant due to fixities
+  redundantInfixBracket :: Map String Fixity -> Int -> a -> a -> Bool
```

Comment:
Why make this a part of the Brackets class given its only used for expressions. I'd suggest moving it to Hint.Fixity, specialising to Expr, and leaving the Brackets module alone.

Reply:
Good question. What I had in mind is that perhaps type operators have fixities too?

Reply:
So do patterns. But I'm not convinced that suggesting more/less fixity based brackets in either of those is a good idea ever. With expressions, everyone pretty much has a good sense that multiplication binds tighter than addition, so its plausible. With types and patterns, there aren't any universally understood fixities, so I think the case is a lot weaker.

Reply:
Alright, removed the polymorphism and it's indeed much simpler now :)